app/lib/katello/resources/registry.rb

```diff
@@ -28,7 +28,7 @@ def load_class
 
             # Pulp 3 has its own registry
             if pulp_primary&.pulp3_repository_type_support?(::Katello::Repository::DOCKER_TYPE)
-              uri = URI(pulp_primary.setting(SmartProxy::PULP3_FEATURE, 'content_app_url'))
+              uri = URI(pulp_primary.pulp_content_url)
               uri.path = "/pulpcore_registry/"
```

Comment:
I think we should create a separate setting for this in smart_proxy_pulp

Reply:
You mean to separate the content app url and the registry url?

Reply:
Today we [expose the content_app_url](https://github.com/theforeman/smart_proxy_pulp/blob/1d4cc85d6b89e292cf613bf0b81ad24e785d4870/lib/smart_proxy_pulp_plugin/pulpcore_plugin.rb#L20). This means Katello doesn't have to guess. I think it would be good if we also added an exposed setting there that is the pulpcore registry URL. We can do that by calculating this value in the Smart Proxy plugin or explicitly setting it in YAML, but Katello shouldn't be responsible for guessing URLs.

Does that clarify it?

Reply:
Oh I get that part. I was asking about whether your comment ultimately meant to not derive the registry URL from "pulp_content_url"

Reply:
Probably. The installer already knows the URL so we can set it in the YAML settings.

Reply:
https://github.com/theforeman/smart_proxy_pulp/pull/30 would be that.